synapse/handlers/presence.py

```diff
@@ -985,7 +986,8 @@ async def user_syncing(
                 await self._update_states(
                     [
                         prev_state.copy_and_replace(
-                            last_user_sync_ts=self.clock.time_msec()
+                            last_active_ts=self.clock.time_msec(),
+                            last_user_sync_ts=self.clock.time_msec(),
                         )
                     ]
                 )
```

Comment:
> I haven't added this code to the worker version of user_syncing as there
> was no code there to set presence status before, but having removed the
> set_state from sync, I'm not certain whether it needs to be there or not,
> so I'd appreciate a check from someone who knows the worker setup better.

We're going to have to update the worker version of `user_syncing` too, otherwise single process synapse and synapse with workers will behave differently.

In worker mode, `self.presence_handler` is a `PresenceHandler` on the worker configured as the presence writer and `WorkerPresenceHandler` on other workers.

eg. on a matrix.org-like setup, `/sync`s would arrive at a worker with a `WorkerPresenceHandler`. `WorkerPresenceHandler.user_syncing` would call `mark_as_coming_online`, which calls `send_user_sync`, which calls `ReplicationCommandHandler.send_user_sync`. The data then arrives on the presence writer at `ReplicationCommandHandler.on_USER_SYNC`, which calls `PresenceHandler.update_external_syncs_row`. `update_external_syncs_row` looks a lot like `user_syncing` and we should probably update this whole code path too.